src/obr/signac_wrapper/operations.py

```diff
@@ -120,24 +124,68 @@ def basic_eligible(job: Job, operation: str) -> bool:
-        parent_job = project.open_job(id=job.sp().get("parent_id"))
-        return parent_job.doc["state"].get("global", "")
+        with open(job.path + f"/../{parent_id}/signac_job_document.json", "r") as jh:
+            parent_job_dict = json.load(jh)
+            return parent_job_dict["state"].get("global", "")
     return ""
 
 
-def _link_path(base: Path, dst: Path, copy_instead_link: bool):
+def _link_path(base: Path, dst: Path, parent_id: str, copy_instead_link: bool):
     """creates file tree under dst with same folder structure as base but all
     files are relative symlinks
     """
+    # NOTE if copy instead linking is requested we
+    # just copy the full tree and are done
+    if copy_instead_link:
+        if dst.exists():
+            shutil.rmtree(f"{dst}")
+        shutil.copytree(src=f"{base}", dst=f"{dst}", symlinks=False)
+        return
+
     # ensure dst path exists
     check_output(["mkdir", "-p", str(dst)])
-
-    # base path might not be ready atm
     for root, folder, files in os.walk(Path(base)):
         relative_path = Path(root).relative_to(base)
 
+        # NOTE Treat processor folder separately
+        # Dont recurse into processor folders for now since that can
+        # become very costly. Instead we copy all processor folder
+        # entirely and pop the folder from the folder list
+        if "processor0" in folder:
+            for fold in folder:
+                if not fold.startswith("processor"):
+                    continue
+                proc_root, proc_folder, _ = next(os.walk(f"{base}/{fold}"))
+                # NOTE The constant folder is linked
+                # to reduce the resulting folder size
+                trgt_proc_fold = f"{dst}/{fold}"
+                for proc_cont in proc_folder:
+                    if proc_cont == "constant":
+                        check_output(["mkdir", "-p", trgt_proc_fold])
+                        check_output(
+                            [
+                                "ln",
+                                "-s",
+                                # we can use this folder format here because
+                                # we know where the parent job lies relative
+                                # to this one in the workspace
+                                f"../../../{parent_id}/case/{fold}/constant",
+                            ],
+                            cwd=trgt_proc_fold,
+                        )
+                    else:
+                        shutil.copytree(
+                            src=f"{proc_root}/{proc_cont}",
+                            dst=f"{trgt_proc_fold}/{proc_cont}",
+                            symlinks=False,
+                        )
+            # pop all processor folder to aviod recursing
+            pop_idx = [i for i, f in enumerate(folder) if f.startswith("processor")]
+            for i in sorted(pop_idx, reverse=True):
+                del folder[i]
```

Comment:
This could have performance problems. If we `del`the first element, all other elements (or the pointers thereof) would have to be moved one element to the front.

Reply:
The pop_idx list is rather small, usually maximum is in the thousands. But traversing the processors is much more expensive. Also we iterate from the back.